OpenSim/Simulation/Model/ForceSet.cpp

```diff
@@ -328,11 +224,14 @@ void ForceSet::updateActuators()
  */
 const Set<Muscle>& ForceSet::getMuscles() const
 {
+    if (_muscles.getSize() == 0)
+        const_cast<ForceSet*>(this)->updateMuscles();
     return _muscles;
```

Comment:
Why is this necessary now? aren't muscles kept in sync forces all along?

Reply:
I agree, we should avoid this. The sets should be kept in sync when they are modified, not when they are accessed. We could add a virtual `extendAppend...()` function if necessary to allow derived classes to do postprocessing.